### DISCUSSION TITLE
Configure minio in expand mode

### DISCUSSION THREAD

**Discussion by modigithub** [minio/minio#15773] [Q&A]
Hello, I currently have 2 servers that I would like to use as test servers for the time being. I successfully installed and set up MinIo on both. Now I want to connect servers 1 and 2 together.

To set up for the 2 servers I used the following tutorial:
https://techviewleo.com/deploy-minio-storage-server-on-debian/
This worked great with my Debian 11 distribution.

Now I want to switch both together. For this I use the normal documentation:
https://min.io/docs/minio/linux/operations/install-deploy-manage/expand-minio-deployment.html

At the moment Server 1 works normally and alone:
https://minio1.demo.modifax.com:9000/

Server 2 worked. Due to the new configuration it doesn't do that anymore:
https://minio2.demo.modifax.com:9000/

When I start server 2 I get the following error message. Unfortunately, it doesn't say exactly what I'm doing wrong.
![image](https://user-images.githubusercontent.com/108171635/193112943-622742f6-fd2c-4cf0-ace2-6ab0d61a4c72.png)

Here is my minio config:
![image](https://user-images.githubusercontent.com/108171635/193112965-07654867-f6fc-47d5-91d0-90c4b665aebe.png)

Here is the Minio.Service config:
![image](https://user-images.githubusercontent.com/108171635/193113002-88baa86a-3719-4725-8239-9cbe16846e19.png)

Currently I have absolutely no idea what I'm doing wrong. Here is some background information:
I have set the following configuration for my minio:
sudo vim /etc/fstab
![image](https://user-images.githubusercontent.com/108171635/193113050-8f6068f5-00d8-4390-8b1b-50bbc96678f3.png)

I only have 1 disk and one path. Both servers are set up identically. What am I doing wrong?

**Comment by vadmeste** [comment]
@modigithub can you paste `journalctl -u minio` output ?

**Comment by modigithub** [comment]
ah ok thank you for the journalctl command. Now its clear whats missing:
![image](https://user-images.githubusercontent.com/108171635/193261473-f0e15ab8-6f1a-4738-8e30-963cde7315d3.png)

I Added the config
![image](https://user-images.githubusercontent.com/108171635/193261355-506d1374-f9af-41b8-84a3-05f8a9ca0042.png)

But now i got the following error:
![image](https://user-images.githubusercontent.com/108171635/193261716-9f1ed429-2a90-497c-bab1-8729ad701a47.png)

Could it be that the Volumes with https dont work?

**Reply by ravindk89** [reply]
The highlighted line is part of the command startup. You need to focus on the journalctl output for both nodes.

> Now I want to switch both together. For this I use the normal documentation:

You cannot go from a single-drive single-node setup to a distributed multi-node setup in the manner above. Your `MINIO_VOLUMES` states:

`https://minio{1...4}.demo.modifax.com:9000/var/lib/minio`

- You only have two servers, and are starting MinIO where it expects at least 4
- Your original startup command was probably something like `https://minio1.demo.modifax.com:9000/var/lib/minio`
- What you're trying to do is go straight to multi-node from single-node while changing the actual topology of the setup, which isn't supported.

How much data do you currently have stored, and are you using any versioning on that data?

**Reply by modigithub** [reply]
Hello, Thanks for the quick reply. I get an idea of how it all works. I changed my minio config. now it should be nearly correct.
![image](https://user-images.githubusercontent.com/108171635/193318874-0d930ff4-e8d3-4a13-bc51-604bb56814a2.png)


Now I get the following message from the journal which I don't understand. What am I missing here?
![image](https://user-images.githubusercontent.com/108171635/193318894-d70ea9f8-6a4f-4b1c-9e12-b1e84de73c73.png)

I stored nothing. The two cloud server where i have installed minio are only for testing und learning purpose. If everything is correctly defined it should be the demo server. For my real project i need a multi-node setup. So i will configure the server to check what i have to do.
No i never will use versioning.

**Reply by harshavardhana** [reply]
Your input for `--console-address is wrong`

**Reply by modigithub** [reply]
Ok thank you for the hint. I think i am nearly at the end. Now i have the following error:
![image](https://user-images.githubusercontent.com/108171635/193397610-c73022cb-090f-404d-977a-b35a1fb0adbd.png)

Now sever 1 cannot connect to server two. does i have to change the folder config to geht acces from minio?

Thats the config of server 1:
![image](https://user-images.githubusercontent.com/108171635/193397702-d9fd6e34-0f43-49da-ba35-dc2df4f79c85.png)
Thats the config of server 2:
![image](https://user-images.githubusercontent.com/108171635/193397741-011d547d-9fb5-4faa-bda2-d15a293eeef4.png)

I Think the configs are correct. Maybe i have to change anything on the folderpermissions or folder configs?

**Reply by harshavardhana** [reply]
yes you need to run MinIO on both nodes  @modigithub as you can see "connection refused" - make sure port 9000 is enabled from your firewall.

**Reply by modigithub** [reply]
yes but thats an chicken-and-egg problem. If i want to start server 1 i got the following error:
![image](https://user-images.githubusercontent.com/108171635/193412573-b60bf7de-c1d2-49fc-9597-cc5f8eb2f92d.png)
And if i start server 2 i got the following message:
![image](https://user-images.githubusercontent.com/108171635/193412524-ef6daec6-ad20-432a-9050-d2bec7e37449.png)

i saw i can enter the comand simulatously with different tools. Is that the was? Does i have to connect two sesseions from two different servers (1 and 2) and enter the command at once?

Port 9000 is enabled.

**Reply by harshavardhana** [reply]
you need fresh drives for distributed setup @modigithub

**Reply by modigithub** [reply]
ok, isn't that a bit strange? why can't I use the main partition? Do I now have to create a new partition with e.g. sda2 and write 18 gb over it? same with servers 2?
![image](https://user-images.githubusercontent.com/108171635/193422027-9e374f8d-67ca-4299-92b7-fc1833a284df.png)

**Reply by harshavardhana** [reply]
Not main partition it needs to be a clean mount i.e you can't have a previously setup single drive config at `.minio.sys/`  delete this it will become a fresh drive for MinIO. 

https://min.io/docs/minio/linux/operations/install-deploy-manage/deploy-minio-single-node-multi-drive.html read the docs first, this applies for all deployments.

> All drives or paths included in the expanded drive list must exist *and* be empty or freshly formatted for MinIO to start successfully.

**Reply by modigithub** [reply]
I I read the documentation and unfortunately I didn't fully understand it. Otherwise I wouldn't be posting here.

I found that 1st of my problems was that I didn't pass the correct number of locations. I need 2 servers with a total of 4 disks (two per server).
This brings me to a working system. I still have the problem with the root directory. Is that the kind of "clean" architecture you mean?
![image](https://user-images.githubusercontent.com/108171635/193424318-963436f6-ef57-41dc-a20e-5a614fb02dcb.png)

I removed the .minio.sys and also created 2 new folders. How can i enable the access? I changed the chown to minio.

**Reply by modigithub** [reply]
Ok, i have solved the access error. But now the minio service are not able to use the disk on the "root" level.
Server 2:
![image](https://user-images.githubusercontent.com/108171635/193445613-ed6421ce-2077-4a2c-a143-444e11d53cc2.png)
Server 2 config:
![image](https://user-images.githubusercontent.com/108171635/193445648-f336644f-5710-4a5f-a7b7-e4c213c3174d.png)

What does that mean? I changed the mount point to minio
![image](https://user-images.githubusercontent.com/108171635/193446320-8a86e418-990f-4446-a2ad-ec947952d4c6.png)

does the second "Disk" have to be on sr0?

The server is now running but it is waiting for a valid drive. Can you pleas help me to understand where i have to mound the folder or Disk to be valid?

**Comment by modigithub** [comment]
ok, i solved the problem. For anyone having the same problem as me, here are the things to consider:
1. After setting up a simple minio server and wanting to expand it, you have to remove the .minio.sys sys.
2. At least 4 drives are required (as described in the documentation). However, at least each drive on a server must be a partition.
3. Every folder must have the correct permission

Thanks to everyone who helped to work out the solution.

**Reply by klauspost** [reply]
Point 1) is only if you did something wrong, and can likely lead to total data loss, so unless you are prepared to lose all data, never do this.
